Source/Renderer/Texture.js

```diff
@@ -197,6 +197,15 @@ function Texture(options) {
 
   gl.pixelStorei(gl.UNPACK_ALIGNMENT, unpackAlignment);
 
+  if (options.skipColorSpaceConversion) {
+    gl.pixelStorei(gl.UNPACK_COLORSPACE_CONVERSION_WEBGL, gl.NONE);
```

Comment:
There are a few cases where this fix alone is not enough.

It works when `source` is an `HTMLImageElement` - this happens when `fetchImage` is called on an external image and `preferImageBitmap` is false or `ImageBitmap` is not supported. Basically glTFs with external images are ok for now...

But embedded images won't work because they use `ImageBitmap` by default (via `loadImageFromTypedArray`) and according to the specs `UNPACK_COLORSPACE_CONVERSION_WEBGL` is ignored for `ImageBitmap`. See https://www.khronos.org/registry/webgl/specs/latest/1.0/#6.10

Another problematic case is when non-power-of-two images are rescaled on a canvas. `HTMLCanvasElement` doesn't seem to get affected by the `UNPACK_COLORSPACE_CONVERSION_WEBGL` flag either. That happens when a texture has to be rescaled to meet WebGL 1's power-of-two requirements for certain samplers. If you simulate that by removing the if guard around [this code](https://github.com/CesiumGS/cesium/blob/3ca010a77a093b2c6b472e0fe307a7ce5cef18db/Source/Scene/Model.js#L2934-L2952) you'll see the problems show up again.

`ImageBitmap` may be fixable according to the specs. I haven't looked into `HTMLCanvasElement` but it's worth investigating.

![image](https://user-images.githubusercontent.com/915398/122625762-c2ec9380-d074-11eb-8548-ea80ffb6ef72.png)

Reply:
CC https://github.com/CesiumGS/cesium/pull/9627. After that PR is merged this fix won't work for external images either since they will now be `ImageBitmap`.

Reply:
For testing embedded images use the glb version of the model: https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureEncodingTest/glTF-Binary

Reply:
In the [Sandcastle](https://sandcastle.cesium.com/#c=dVTvT9swEP1XrH4hSJmTjp8tBQ0B0yYNjbGKL8s+mOTSWHPsyj4Xuqn/+85J04YC/uTcvXv37tmOAmQI1gqp2Tm7Aid9zXMLAuEaSuEVTtvsnTULWYB9kPB0awpQLto/y3SmFTEsKAiWCDQ8dSQPTSzay5vPK6ORWMDuxexfphktBwpylEZ/1YXMBRrLxqwUykHcAvC9zuMuFfeZoHhP6gb/a/i7K6mMV8WlljVNSnnrIdOrdqLS60YXa31oKCJvVcwqkLMK97sJ2rE5aJQowXELtVnApVJrawIm2DM3TjaEG4evhEXaCX3AS2vqa5hZABcND0b8eDQ8SY9jdnDEj09Hh6cnm7ZnW8YKRCH1bEt4K7DiaO4pLLSLRmkfPZeYV4RN+VEvao1SIbgO9cjnO0f5pe12F2juqSpat49b5rih6jc0VpIp4uXMU0vKSmNrx6sdwh+efLaa4FHnVRxkvLSx8XlJhLu+i6KI1kcS1sbu8WYbb7N9beP+Vw9ThyOnbI80LG8lBcNNeBl3uVBAmWGapjzd5lbtdtVZs9ZdquXURO009BpY4W0jYMwOCcuxAh11dzDa74tYE6AV+R8objo/WqqzXrdV9zJJLAGyQYU4d+MkyZ8/pnwmsfKPXJpkprD8QDcRE/T0AKVQ7QtLpvCM3sKNzk04qWmAzNT081sJHliyQXNWr57MEXkSFA3iwcThUsFFN84nWc+NbSRGnCcI9VxRrUsePU2HPHeuM26S9EsnhVwwWZxng51/SzZguRLOUab0Sv2UfyEbXEwSwr8qVaa5gt8XYJVYBlg1vPjWBjnnk4Q+365EY9SjsDvM/wE) I'm using to test this, how do I test the glb version of the model? If I replace the glTF URL with the .glb one, it gives me an error, even if I add the `?raw=true` flag at the end of the URL to get the raw file.

Reply:
Also, it seems that there is a flag in `ImageBitmapOptions` that is equivalent to `UNPACK_COLORSPACE_CONVERSION_WEBGL`, but it will require me modifying functions like `createImageBitmapFromBlob` and `loadImageFromTypedArray`, is that ok?

Reply:
> it will require me modifying functions like createImageBitmapFromBlob and loadImageFromTypedArray, is that ok?

Yup that's the right approach. I left a comment with more detail about that.

Reply:
Curious, did you have a chance to look into into the `HTMLCanvasElement` problem?

Reply:
Never mind, I saw your comment below.

Reply:
Hm, I'm unable to reproduce the HTMLCanvasElement problem. I removed the `if` guards around the code that you linked to, but my result was the same whether or not they were there.

Reply:
I tried to look into this anyway, and unfortunately, the image handling functions in [`CanvasContextRendering2D`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D) have no options for handling color profiles. There's a [proposal](https://wiki.whatwg.org/wiki/CanvasColorSpace) to add a feature that specifies a color profile for a context, but nothing actually official.

Reply:
Ok thanks for looking into it. This is a less common case so I'm ok with moving forward with the PR.